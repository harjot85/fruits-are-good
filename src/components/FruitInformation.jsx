import React from "react";
import {ListGroup, ListGroupItem } from "reactstrap";

const FruitInformation = (props) => {
    const { fruit } = props;

    console.log(fruit);
    return (
        <>
            <ListGroup>
            <ListGroupItem>Name: {fruit.name}</ListGroupItem>
                <ListGroupItem>Family: {fruit.family}</ListGroupItem>
                <ListGroupItem>
                    Calories: {fruit.nutritions.calories}
                </ListGroupItem>
            </ListGroup>
        </>
    );
};

export default FruitInformation;
