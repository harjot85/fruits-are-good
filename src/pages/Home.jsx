import React, { Component } from "react";
import FruitForm from "../components/FruitForm";
import FruitInformation from "../components/FruitInformation";
import { Row, Col, Container, Alert } from "reactstrap";
import { getLocalFruitDataByFruitName } from "../api/getFruitInfo";

class Home extends Component {
    state = {
        fruit: "",
        fruitData: {},
        errorState: false,
        error: "",
        isLoading: false,
    };

    onClickHandler = () => {
        this.getData();
    };

    getData() {
        const { fruit } = this.state;

        if (fruit) {
            const fruitData = getLocalFruitDataByFruitName(fruit);

            if (fruitData === undefined) {
                this.setState({ errorState: true });
            } else {
                this.setState({ fruitData });
            }
        }
    }

    onInputChange = (e) => {
        const fruit = e.target.value;
        this.setState({
            fruit,
            errorState: false,
            fruitData: {},
        });
    };

    render() {
        console.log("error ", this.state.errorState);
        return (
            <>
                <div style={{ textAlign: "center", margin: "25px 0" }}>
                    <h1>Awesome fruits! </h1>
                </div>
                <Row>
                    <Col>
                        {console.log(this.state.fruit)}
                        <FruitForm
                            onInputChange={this.onInputChange}
                            onClickHandler={this.onClickHandler}
                            onKeyDown={this.onKeyDown}
                        />
                    </Col>
                </Row>

                <Container className="themed-container">
                    {this.state.errorState ? (
                        <Alert color="danger">
                            <div style={{ textAlign: "center" }}>
                                <p>Fruit not found</p>
                            </div>
                        </Alert>
                    ) : (
                        Object.keys(this.state.fruitData).length > 0 && (
                            <Alert color="warning" style={{ padding: "1rem" }}>
                                <FruitInformation
                                    fruit={this.state.fruitData}
                                />
                            </Alert>
                        )
                    )}
                </Container>
            </>
        );
    }
}

export default Home;
