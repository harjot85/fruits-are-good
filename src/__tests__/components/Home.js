import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import FruitInformation from "../../components/FruitInformation";

// UI Components
import Home from "../../pages/Home";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Home />);
});

describe("<Home />", () => {
    it("should match snapshot with props", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should not render <FruitInformation /> errorState is true", () => {
        const wrapper = shallow(<Home />);
        wrapper.setState({ errorState: true });
        expect(wrapper.find(FruitInformation)).toHaveLength(0);
    });
});
