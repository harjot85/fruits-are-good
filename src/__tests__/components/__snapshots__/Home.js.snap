// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot with props 1`] = `ShallowWrapper {}`;
